daemon/qrexec-client.c

```diff
@@ -757,26 +771,28 @@ int main(int argc, char **argv)
     }
 
     if (strcmp(domname, "dom0") == 0 || strcmp(domname, "@adminvm") == 0) {
+        assert(src_domain_name);
+        if (src_domain_name == NULL)
+            abort();
```

Comment:
Why `assert` and `if/abort`?

Reply:
If `NDEBUG` is defined, `assert()` is no-op, so it can't be used for assuring correctness - but still produce more useful message for debugging.

Reply:
Sure, but then just add a `printf` instead of adding another code path with a separate check. Or maybe even add an abort that takes a message (keep in mind that in other places abort might be called with stdout/err connected to something special, so this should be opt-in).